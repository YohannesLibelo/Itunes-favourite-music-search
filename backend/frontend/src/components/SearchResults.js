import React from 'react';

const SearchResults = ({ results, likedItems, onToggleFavorite }) => {
  const handleToggleFavorite = (item) => {
    onToggleFavorite(item);
  };

  return (
    <div>
   
      {results.map((result) => (
        <div key={result.trackId}>
          <h3>{result.trackName}</h3>
          <p>Artist: {result.artistName}</p>
          {/* Display other relevant information */}
          <button onClick={() => handleToggleFavorite(result)}>
            {likedItems.some((likedItem) => likedItem.trackId === result.trackId) ? '‚ù§Ô∏è' : 'ü§ç'}
          </button>
        </div>
      ))}
    </div>
  );
};

export default SearchResults;
