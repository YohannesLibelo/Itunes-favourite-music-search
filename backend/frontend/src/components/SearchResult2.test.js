
//snapshot test for the SearchResults component using React Testing Library 
import React from 'react';
import { render } from '@testing-library/react';
import SearchResults from './SearchResults';

describe('SearchResults component', () => {
  test('renders correctly', () => {
    // Arrange
    //Define the results array with two items.
    const results = [
      {
        trackId: 1,
        trackName: 'Track 1',
        artistName: 'Artist 1',
      },
      {
        trackId: 2,
        trackName: 'Track 2',
        artistName: 'Artist 2',
      },
    ];
    //Initialize the likedItems array as empty.
    const likedItems = [];
    //Create a mock function called onToggleFavorite using jest.fn().

    const onToggleFavorite = jest.fn();

    // Act
    const { container } = render(
      <SearchResults results={results} likedItems={likedItems} onToggleFavorite={onToggleFavorite} />
    );

    // Assert
    expect(container).toMatchSnapshot();
  });
});
