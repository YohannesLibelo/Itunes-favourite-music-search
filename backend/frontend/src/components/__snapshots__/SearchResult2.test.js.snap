// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchResults component renders correctly 1`] = `
<div>
  <div>
    <div>
      <h3>
        Track 1
      </h3>
      <p>
        Artist: 
        Artist 1
      </p>
      <button>
        🤍
      </button>
    </div>
    <div>
      <h3>
        Track 2
      </h3>
      <p>
        Artist: 
        Artist 2
      </p>
      <button>
        🤍
      </button>
    </div>
  </div>
</div>
`;
